src/libs/ajax/AzureStorage.ts

```diff
@@ -64,21 +79,29 @@ export const AzureStorage = signal => ({
     const xml = new window.DOMParser().parseFromString(text, 'text/xml')
     const blobs = _.map(
       blob => ({
-        name: _.head(blob.getElementsByTagName('Name')).textContent,
+        name: _.head(blob.getElementsByTagName('Name'))?.textContent || '',
         lastModified: new Date(
-          _.head(blob.getElementsByTagName('Last-Modified')).textContent
+          _.head(blob.getElementsByTagName('Last-Modified'))?.textContent || '0'
         ).getTime()
       }),
       xml.getElementsByTagName('Blob')
-    )
+    ) as AzureFileRaw[]
 
     const filteredBlobs = _.filter(blob => _.endsWith(suffixFilter, blob.name), blobs)
     return filteredBlobs
   },
 
-  listNotebooks: async workspaceId => {
+  listNotebooks: async (workspaceId: string): Promise<AnalysisFile[]> => {
     const notebooks = await AzureStorage(signal).listFiles(workspaceId, '.ipynb')
-    return _.map(notebook => ({ ...notebook, application: tools.Jupyter.label }), notebooks)
+    return _.map(notebook => ({
+      ...notebook,
+      name: notebook.name as AbsolutePath,
+      ext: getExtension(notebook.name),
+      displayName: getDisplayName(notebook.name),
+      fileName: getFileName(notebook.name),
+      tool: toolLabelTypes[tools.Jupyter.label],
+      cloudProvider: cloudProviderTypes[cloudProviders.azure.label]
```

Comment:
@msilva-broad is this valuable vs casting?

Reply:
hmm... can you be a bit more specific?

Reply:
ok, I see what you are doing here...  cloudProviders is defined in a js file as:

export const cloudProviders = {
  azure: { label: 'AZURE', iconTitle: 'Microsoft Azure' },
  gcp: { label: 'GCP', iconTitle: 'Google Cloud Platform' }
}

So I suggest two things:
1) change usages here (and in similar spots elsewhere in the PR) to the pattern:
    cloudProvider: cloudProviderTypes.AZURE
2) change cloudProviders export to itself has good typing (including using CloudProviderType as type of label sub-property)   - this could be in a follow-up PR